### DISCUSSION TITLE
"[!] No Steam API DLL was found in the game!"

### DISCUSSION THREAD

**Discussion by misteryora** [BigBoiCJ/SteamAutoCracker#24] [Q&A] [answered]
trying this program for the first time i downloaded clean file of a game (in this case Battleblock Theater cause it's rather small) from cs.rin.ru and put it in my "pirated" folder but when i run the program after saying "couldn't find folder ..." i typed the game folder name, the program said 
"found game folder ..." 
"Searching Steam API DLLs and cracking them"
"Finished cracking the game"
**[!] No Steam API DLL was found in the game!**

and the game won't open

after that i tried putting the files i downloaded on my "Steam" folder, tried it again, same input, and it works.. what am i doing wrong when i put the file in the pirated folder? is it because i downloaded a clean steam files that it only works if i put it in my steam folder with other legit games i got? is there a way that i can do it in pirated folder instead because i don't really want to put my legit and non bought games in the same folder

**Comment by BigBoiCJ** [comment, marked as answer]
Hmm, that's weird indeed, I'll try reproducing what you've reported me.
The actual path used shouldn't matter as the only things that are used are inside the game's folder (here, steam_api DLLs)

Might be caused by an incorrect configuration of the pirated games path.
You should configure the program so the pirated games path points to a folder where you'll have multiple games folders directly inside. The Steam equivalent folder is `../SteamApps/common/`

--- UPDATE ---

After testing, it worked correctly for me. My hypothesis right now is that you've set-up a wrong pirated games location.
You can manually edit `config.ini` and fix the pirated line.

**IT IS REALLY IMPORTANT THAT YOUR PATH IS FORMATED IN A SIMILAR WAY, WITH FORWARD SLASHES `/` AND A FORWARD SLASH AT THE END**
This is an example path: `pirated = D:/Games/Crack/` 
**or you can delete config.ini and restart SAC. This will allow you to configure SAC again through the program.**

I will replace my actual pirated games path by `D:/Games/Crack/` in the logs below:

```
Steam Auto Cracker (v1.2.9) by BigBoiCJ
-----
Enter the Name or AppID of the game you want to Crack: BattleBlock Theater
- Importing and searching the App List, this could take a few seconds if your computer isn't powerful enough.

[1/4] Retrieving game informations from Steam...
- Game found! Name: BattleBlock Theater® - AppID: 238460

[2/4] Retrieving DLCs...
- No DLC found for this game!

[3/4] Searching the game folder...
Couldn't automatically find the folder name of the game
Please write the folder name (in SteamApps/common/ or in your pirated games folder)
Folder name: BattleBlock Theater
- Found the game folder on your computer: D:/Games/Crack/BattleBlock Theater

[4/4] Searching Steam API DLLs and cracking them...
- Attempting to run Steamless on BattleBlockTheater.exe

[[[ Steamless logs ]]]
  _________ __                        .__
 /   _____//  |_  ____ _____    _____ |  |   ____   ______ ______
 \_____  \\   __\/ __ \\__  \  /     \|  | _/ __ \ /  ___//  ___/
 /        \|  | \  ___/ / __ \|  Y Y  \  |_\  ___/ \___ \ \___ \
/_______  /|__|  \___  >____  /__|_|  /____/\___  >____  >____  >
        \/           \/     \/      \/          \/     \/     \/

Steamless - SteamStub DRM Remover
by atom0s

GitHub    : https://github.com/atom0s/Steamless
Homepage  : https://atom0s.com
Donations : https://paypal.me/atom0s
Donations : https://github.com/sponsors/atom0s
Donations : https://patreon.com/atom0s

[ExamplePlugin] ExamplePlugin was initialized!
[Steamless] Loaded plugin: Example Plugin - by Steamless Development Team (v.1.0.0.0)
[Steamless] Loaded plugin: SteamStub Variant 1.0 Unpacker (x86) - by atom0s (v.1.0.0.0)
[Steamless] Loaded plugin: SteamStub Variant 2.0 Unpacker (x86) - by atom0s (v.1.0.0.3)
[Steamless] Loaded plugin: SteamStub Variant 2.1 Unpacker (x86) - by atom0s (v.1.0.0.8)
[Steamless] Loaded plugin: SteamStub Variant 3.0 Unpacker (x64) - by atom0s (v.1.0.0.3)
[Steamless] Loaded plugin: SteamStub Variant 3.0 Unpacker (x86) - by atom0s (v.1.0.0.1)
[Steamless] Loaded plugin: SteamStub Variant 3.1 Unpacker (x86) - by atom0s (v.1.0.0.2)
[Steamless] Loaded plugin: SteamStub Variant 3.1.x Unpacker (x64) - by atom0s (v.1.0.0.3)
[ExamplePlugin] ExamplePlugin was asked to check if it can process a file!
[Steamless] All unpackers failed to unpack file.
[[[ -------------- ]]]

- Couldn't run Steamless on BattleBlockTheater.exe, it is probably not under DRM.
- Found Steam API DLL(s) in D:/Games/Crack/BattleBlock Theater and cracked them successfully

-----
Finished cracking the game!
```

**Reply by BigBoiCJ** [reply]
Here is a visualization of game files:

D:/Games/Crack/ contains:
- Folder: BattleBlock Theater
- Other game folders

D:/Games/Crack/BattleBlock Theater/ contains:
- Folder: GameData
- File: BattleBlockTheater.exe
- File: bink2w32.dll
- File: steam_api.dll
- File: steam_appid.txt

**Version used: Build 842731 - Uploaded by Rui on cs.rin.ru (Steam Content Sharing)**

**Reply by misteryora** [reply]
ah the forward slashes did the trick! turns out for my first try i was editing it the path in config.ini using copied path with anti-slashes but then later when i tried it with my steam folder i first deleted my config.ini, started the program, and pasting the folder path in the program in which then corrected the anti-slashes automatically, that's why it worked.

It is kinda weird that it still runs and says "finished cracking the game" even though it actually met an "error" but still thank you very much for both the help and the program!

**Reply by BigBoiCJ** [reply]
Yea indeed this can be weird, it just shows that when the program finishes without any error that makes it crash.
Since having no steam api DLL doesn't crash the program and just displays a warning, it still counts as if the game was cracked properly, but yea I might have to change that to make it so that it doesn't display this message if no API DLL is found and if Steamless did nothing, because if that happens then SAC did nothing.

I'm glad we were able to figure this out! :)
